R/OWS-methods.R

```diff
@@ -214,27 +214,24 @@ setMethod(f = "checkRequest",
 #   (space)             %20
 #
 .kvpEscapeSpecialCharacters <- function(x) {
-  .escaped <- gsub(x = x, pattern = ":", replacement = "%3A")
-  #.escaped <- gsub(.escaped, pattern = "/", replacement = "%2F")
-  .escaped <- gsub(x = .escaped, pattern = "#", replacement = "%23")
-  .escaped <- gsub(x = .escaped, pattern = "\\?", replacement = "%3F")
-  .escaped <- gsub(x = .escaped, pattern = "=", replacement = "%3D")
-  .escaped <- gsub(x = .escaped, pattern = "&", replacement = "%26")
-  .escaped <- gsub(x = .escaped, pattern = ",", replacement = "%2C")
-  .escaped <- gsub(x = .escaped, pattern = "\\+", replacement = "%2B")
-  .escaped <- gsub(x = .escaped, pattern = "@", replacement = "%40")
-  .escaped <- gsub(x = .escaped, pattern = " ", replacement = "%20")
-  #.escaped <- gsub(x = .escaped, pattern = "\"", replacement = "%22")
-  return(.escaped)
+  # handle S4 objects
+  if (is.list(x) && length(x) == 1 && isS4(x[[1]])) {
+    x <- x[[1]]
+  }
+  if (isS4(x)) {
+    x <- toKVPString(x)
```

Comment:
Where does the function `toKVPString` come from??

Reply:
Good question. I can not find it anywhere, too. It seems to be an left open end. The idea behind this could have been to add a method for every class (where it's needed) a method to get a string representation that could be encoded as an KVP request.
I could remove it and add a stop call because the method is currently not able to convert S4 objects.

What do you think?

Reply:
Yes, please `stop` for S4 objects.

Reply:
Yes, please `stop` for S4 objects.

Reply:
Will fix myself.